Classes/FrontEnd/DefaultController.php

```diff
@@ -3047,9 +3150,27 @@ protected function createRegistrationForm()
             $registrationEditor->setRegistration($this->registration);
         }
 
+        foreach ($this->getRegistrationViewHooks() as $hook) {
+            $hook->modifyRegistrationForm($this, $registrationEditor);
+        }
+
         return $registrationEditor->render();
     }
 
+    /**
+     * Creates the registration page footer.
+     *
+     * @return string HTML code including the title and error message
+     */
+    protected function createRegistrationFooter()
```

Comment:
Please let's move this refactoring (without the hook at first) into a separate PR.

Reply:
Resolved by #309